File: 7/php/euler7.php
<?php

function isPrime (int $x): bool
{
    if ($x < 2) {
        return false;
    }

    $divisor = 2;
	while ($divisor * $divisor <= $x) {
        if (!($x % $divisor)) {
            return false;
        }
        $divisor += 1;
    }

	return true;
}

function countPrimes (int $n)
{
    $x = 2;
	for(;$n; $x += 1) {
        if (isPrime($x)) {
            $n -= 1;
        }
    }

	return $x-1;
}

function bench(callable $fn): array
{
    $startTime = microtime(true);
    $res = $fn();
    $endTime = microtime(true);

    return [$endTime - $startTime, $res];
}

$n = $_SERVER['argv'][1];

[$spent, $prime] = bench(function () use ($n){
    return countPrimes($n);
});

print "Prime: $prime\n";
print "Time: $spent\n";

File: 7/python/euler7.py
#!/usr/bin/python

import sys
import time


def isPrime(x) :
    divisor = 2

    if x < 2 :
        return False;

    while divisor * divisor <= x :
        if x % divisor == 0 :
            return False;
        divisor += 1;

    return True;

def countPrimes (n) :
    x = 2
    while n :
        if isPrime(x) :
            n -= 1;
        x += 1

    return x - 1



if __name__ == "__main__":
    n = int(sys.argv[1])
    start = time.time()
    x = countPrimes(n)
    end = time.time()

    print (n," prime = ", x)
    print (end - start)
